src/test/java/com/pubnub/contract/subscribe/steps/GivenSteps.kt

```diff
@@ -0,0 +1,76 @@
+package com.pubnub.contract.subscribe.steps
+
+import com.pubnub.api.PubNub
+import com.pubnub.api.callbacks.SubscribeCallback
+import com.pubnub.api.models.consumer.PNStatus
+import com.pubnub.api.models.consumer.objects_api.channel.PNChannelMetadataResult
+import com.pubnub.api.models.consumer.objects_api.membership.PNMembershipResult
+import com.pubnub.api.models.consumer.objects_api.uuid.PNUUIDMetadataResult
+import com.pubnub.api.models.consumer.pubsub.PNMessageResult
+import com.pubnub.api.models.consumer.pubsub.PNPresenceEventResult
+import com.pubnub.api.models.consumer.pubsub.PNSignalResult
+import com.pubnub.api.models.consumer.pubsub.files.PNFileEventResult
+import com.pubnub.api.models.consumer.pubsub.message_actions.PNMessageActionResult
+import com.pubnub.contract.state.World
+import com.pubnub.contract.subscribe.state.SubscribeState
+import io.cucumber.java.en.Given
+import java.util.*
+import java.util.concurrent.CountDownLatch
+
+
+class GivenSteps(
+    private val world: World,
+    private val subscribeState: SubscribeState
+) {
+
+
+    @Given("I subscribe to {string} channel")
+    fun I_subscribe_to_the_channel(channelName: String) {
+        subscribeState.messageDeliveredToListener = CountDownLatch(2)
```

Comment:
Why 2 here? There's nothing here that would say anything about it. What if someone decides to use the same step in scenario with just 1 message?

Reply:
I'm not entirely sure that countdown latch is the way to go here. We don't know anything what should be done at this point of time. This step (which is very very generic) should do something very generic

Reply:
ok. I will introduce changes.